[PLUGIN]: Initial FastAPI Integration

amadolid opened the pull request.

amadolid committed:
[PLUGIN]: Initial FastAPI Integration

amadolid committed:
[MYPY]: 54/220 warnings addressed

amadolid committed:
[MYPY]: all warnings addressed

amadolid committed:
[UNIT-TEST]: Add Common Test including crud operations

amadolid committed:
[MYPY]: motor-types alignments

amadolid committed:
[MINOR]: Chang return type of User.model function

amadolid committed:
[MINOR]: Support all optional params on all database operations

ypkang commented:
node with <init> seems to break.

```
  File "myca.jac", line 437, in go_to_user
    users = await [-->];
             ^^^^^^^^^^^
  File "/Users/yipingkang/projects/jaclang/examples/plugins/jaclang_fastapi/jaclang_fastapi/plugins/graph_doc.py", line 88, in edge_ref
    await node._jac_.edges_to_nodes(
  File "/Users/yipingkang/projects/jaclang/examples/plugins/jaclang_fastapi/jaclang_fastapi/plugins/common.py", line 542, in edges_to_nodes
    e._jac_.target = t = await t.connect()
                         ^^^^^^^^^^^^^^^^^
  File "/Users/yipingkang/projects/jaclang/examples/plugins/jaclang_fastapi/jaclang_fastapi/plugins/common.py", line 243, in connect
    data = self.arch = await cls.Collection.find_by_id(self.id)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/yipingkang/projects/jaclang/examples/plugins/jaclang_fastapi/jaclang_fastapi/collections/base.py", line 207, in find_by_id
    return await cls.find_one({"_id": ObjectId(id)}, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/yipingkang/projects/jaclang/examples/plugins/jaclang_fastapi/jaclang_fastapi/collections/base.py", line 199, in find_one
    return cls.__document__(result)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/yipingkang/projects/jaclang/examples/plugins/jaclang_fastapi/jaclang_fastapi/plugins/common.py", line 374, in __document__
    return cls.build_node(
           ^^^^^^^^^^^^^^^
  File "/Users/yipingkang/projects/jaclang/examples/plugins/jaclang_fastapi/jaclang_fastapi/plugins/common.py", line 48, in build_node
    arch: NodeArchitype = doc_anc.build(**(doc.get("context") or {}))
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/yipingkang/projects/jaclang/examples/plugins/jaclang_fastapi/jaclang_fastapi/plugins/common.py", line 208, in build
    self.arch = self.class_ref()(**kwargs)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/yipingkang/projects/jaclang/jaclang/plugin/default.py", line 83, in new_init
    inner_init(self, *args, **kwargs)
TypeError: user_root.__init__() got an unexpected keyword argument 'jid'
```

amadolid committed:
[MINOR]: Implement generic type for basecollection

amadolid committed:
[MINOR]: Improvements

CHANGES:
 - Support non API walker
 - 2 query (get edges, get nodes) per current node visit
 - Utilize Generic typing for proper return type hints

amadolid committed:
[SECURITY]: Integrate access validation with level

CHANGES:
 - Added unit test
